core/chains/evm/client/errors.go

```diff
@@ -100,6 +100,22 @@ var besu = ClientErrors{
 	Fatal:                             besuFatal,
 }
 
+// Erigon
+// See: https://github.com/ledgerwatch/erigon/blob/devel/core/tx_pool.go
+//      https://github.com/ledgerwatch/erigon/blob/devel/core/error.go
+//      https://github.com/ledgerwatch/erigon/blob/devel/core/vm/errors.go
+// Note: some error definitions are unused, many errors are created inline.
+var erigonFatal = regexp.MustCompile(`(: |^)(exceeds block gas limit|invalid sender|negative value|oversized data|gas uint64 overflow|intrinsic gas too low|nonce too high)$`)
+var erigon = ClientErrors{
+	NonceTooLow:                       regexp.MustCompile(`(: |^)nonce too low$`),
+	ReplacementTransactionUnderpriced: regexp.MustCompile(`(: |^)replacement transaction underpriced$`),
+	TransactionAlreadyInMempool:       regexp.MustCompile(`(: |^)(block already known|already known)`),
+	TerminallyUnderpriced:             regexp.MustCompile(`(: |^)transaction underpriced$`),
+	InsufficientEth:                   regexp.MustCompile(`(: |^)(insufficient funds for transfer|insufficient funds for gas \* price \+ value|insufficient balance for transfer)$`),
+	TxFeeExceedsCap:                   regexp.MustCompile(`(: |^)tx fee \([0-9\.]+ [a-zA-Z]+\) exceeds the configured cap \([0-9\.]+ [a-zA-Z]+\)$`),
+	Fatal:                             erigonFatal,
+}
```

Comment:
If every error is identical to go-ethereum, it ought not to be necessary to specify this block.

Reply:
Some differences I can see from the `geth` defined errors are:

- `TransactionAlreadyInMempool` covers `block already known`, which `geth` doesn't.

So I understand this would be the only error that would exist in this specific section.

cc: @pinebit

Reply:
As we continue to add more clients, it is increasingly important to consider that we match all regexs, not just a subset for a particular client. Both to keep things simple, but also for efficiency. (maybe worth parallizing these at some point?)